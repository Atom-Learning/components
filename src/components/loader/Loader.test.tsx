import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import React from 'react'

import { Loader } from '.'

describe(`Loader component`, () => {
  it('renders a loader', async () => {
    const message = 'Content is loading'
    const { container } = render(<Loader message={message} />)

    await screen.getByRole('alert')
    await screen.getByText(message)

    expect(container).toMatchSnapshot()
    expect(await axe(container)).toHaveNoViolations()
  })
})
