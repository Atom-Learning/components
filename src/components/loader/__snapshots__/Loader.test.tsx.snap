// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders a loader 1`] = `
@keyframes sxwho2l {
  0% {
    opacity: 0;
  }

  30% {
    opacity: 1;
  }

  50% {
    opacity: 1;
  }

  80% {
    opacity: 0;
  }

  100% {
    opacity: 0;
  }
}

.sx2cs53 {
  display: flex;
}

.sx2cs53uiywd--css {
  justify-content: center;
  white-space: nowrap;
}

.sx0yv19 {
  animation-name: sxwho2l;
  animation-duration: 1s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
  background-color: currentColor;
  border-radius: 100%;
}

.sx0yv19:nth-child(1) {
  animation-delay: -300ms;
}

.sx0yv19:nth-child(2) {
  animation-delay: -150ms;
}

.sx0yv19:nth-child(3) {
  animation-delay: 0;
}

.sx0yv19plvfs--size-md {
  height: 8px;
  margin-left: 3px;
  margin-right: 3px;
  width: 8px;
}

<div>
  <div
    aria-live="assertive"
    class="sx2cs53 sx2cs53uiywd--css"
    role="alert"
  >
    <span
      data-radix-visually-hidden=""
      style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
    >
      Content is loading
    </span>
    <div
      class="sx0yv19 sx0yv19plvfs--size-md"
    />
     
    <div
      class="sx0yv19 sx0yv19plvfs--size-md"
    />
     
    <div
      class="sx0yv19 sx0yv19plvfs--size-md"
    />
  </div>
</div>
`;
